### DISCUSSION TITLE
Mineflayer bot keeps giving me play.toServer error

### DISCUSSION THREAD

**Discussion by Haasin-azeem** [PrismarineJS/mineflayer#3149] [Q&A]
my code is 
`
function sell() {
  bot.chat("/ ds qsell") // opens sell window
  bot.on("windowOpen", (window) => {
    window.deposit(859, null, 64) // deposits blaze rod intosell window
    window.close()
  })
}
`

It keeps giving me the error 
`
TypeError: Serialization error for play.toServer : SizeOf error for undefined : Cannot read properties of undefined (reading 'present')at Object.slot (eval at compile (C:\Users\Administrator\Desktop\pw_pretz\node_modules\protodef\src\compiler.js:258:12), <anonymous>:51:27)at eval (eval at compile (C:\Users\Administrator\Desktop\pw_pretz\node_modules\protodef\src\compiler.js:258:12), <anonymous>:691:29)at eval (eval at compile (C:\Users\Administrator\Desktop\pw_pretz\node_modules\protodef\src\compiler.js:258:12), <anonymous>:693:11)at Object.packet_window_click (eval at compile (C:\Users\Administrator\Desktop\pw_pretz\node_modules\protodef\src\compiler.js:258:12), <anonymous>:696:9)at eval (eval at compile (C:\Users\Administrator\Desktop\pw_pretz\node_modules\protodef\src\compiler.js:258:12), <anonymous>:1036:64)at packet (eval at compile (C:\Users\Administrator\Desktop\pw_pretz\node_modules\protodef\src\compiler.js:258:12), <anonymous>:1069:9)at CompiledProtodef.sizeOf (C:\Users\Administrator\Desktop\pw_pretz\node_modules\protodef\src\compiler.js:89:14)at e.message (C:\Users\Administrator\Desktop\pw_pretz\node_modules\protodef\src\compiler.js:96:40)at tryCatch (C:\Users\Administrator\Desktop\pw_pretz\node_modules\protodef\src\utils.js:50:16)at CompiledProtodef.createPacketBuffer (C:\Users\Administrator\Desktop\pw_pretz\node_modules\protodef\src\compiler.js:96:20) {field: 'play.toServer'}
`
even though i have specified version "1.18.2" 

`
var settings = {
    username: "pw_pretz",
    host: "herobrine.org",
    version: "1.18.2"
};
`

**Comment by PondWader** [comment]
Update Mineflayer

**Reply by pwpretz** [reply]
ok thank you it worked
